src/drivers/batt_smbus/batt_smbus.cpp

```diff
@@ -974,40 +880,37 @@ batt_smbus_main(int argc, char *argv[])
 
 		default:
 			batt_smbus_usage();
-			exit(0);
+			return 0;
 		}
 	}
 
 	if (optind >= argc) {
 		batt_smbus_usage();
-		exit(1);
+		return 1;
 	}
 
 	const char *verb = argv[optind];
 
 	if (!strcmp(verb, "start")) {
 		if (g_batt_smbus != nullptr) {
-			PX4_ERR("already started");
-			return 1;
+			errx(1, "already started");
```

Comment:
Can you remove anything that actually exists? For example replace errx with PX4_ERR and a return. Then this driver will be usable out of the box on linux where an exit actually kills the px4 process.

Reply:
Done